homeassistant/components/calendar/__init__.py

```diff
@@ -146,6 +176,56 @@ def cleanup(self):
             'description': None
         }
 
+        self._events = None
+
+    def normalize_event(self, event, state_event=False):
+        """Normalize a calendar event."""
+        start = get_date(event['start'])
+        end = get_date(event['end'])
+
+        start_formatted = start.strftime(
+            DATE_STR_FORMAT) if start is not None else None
+        end_formatted = end.strftime(
+            DATE_STR_FORMAT) if end is not None else None
+
+        normalized_event = {}
+        summary = event.get('summary', '')
+
+        # check if we have an offset tag in the message
+        # time is HH:MM or MM
+        if state_event:
+            offset_time = 0
+            reg = '{}([+-]?[0-9]{{0,2}}(:[0-9]{{0,2}})?)'.format(self._offset)
+            search = re.search(reg, summary)
+            if search and search.group(1):
+                time = search.group(1)
+                if ':' not in time:
+                    if time[0] == '+' or time[0] == '-':
+                        time = '{}0:{}'.format(time[0], time[1:])
+                    else:
+                        time = '0:{}'.format(time)
+
+                offset_time = time_period_str(time)
+                summary = (summary[:search.start()] + summary[search.end():]) \
+                    .strip()
+            else:
+                offset_time = dt.dt.timedelta()  # default it
+
+            normalized_event['offset_time'] = offset_time
+            normalized_event['dt_start'] = start
+            normalized_event['dt_end'] = end
+
+        # cleanup the string so we don't have a bunch of double+ spaces
+        normalized_event['message'] = re.sub('  +', '', summary).strip()
+        normalized_event['location'] = event.get('location', '')
+        normalized_event['description'] = event.get('description', '')
+        normalized_event['link'] = event.get('htmlLink', '')
```

Comment:
I can understand that we want things to be strings. But why should things change name from the vCARD spec? The moment we change names, we make things harder.